runtime/runtime/src/lib.rs

```diff
@@ -318,15 +325,15 @@ impl Runtime {
     /// In case of an error, returns either `InvalidTxError` if the transaction verification failed
     /// or a `StorageError` wrapped into `RuntimeError`.
     #[instrument(target = "runtime", level = "debug", "process_transaction", skip_all, fields(
-        tx_hash = %signed_transaction.get_hash(),
+        tx_hash = %validated_tx.to_signed_tx().get_hash(),
```

Comment:
This pattern occurs multiple times throughout this PR, so would make sense to add a method for this.